lib/ar_transaction_changes.rb

```diff
@@ -34,6 +34,11 @@ def write_attribute(attr_name, value)
     end
   end
 
+  def attribute_will_change!(attr_name)
+    transaction_changed_attributes[attr_name] = _read_attribute_for_transaction(attr_name)
```

Comment:
transaction_changed_attributes is meant to store the original value, so the key shouldn't be set if it already exists.  Also, this isn't handling serialized attributes in a way consistent with `_store_transaction_changed_attributes`.

I've just pushed a commit to fix both of these issues.